src/forms/common-components/AuthenticatedRedirection.jsx

```diff
@@ -41,7 +44,21 @@ const AuthenticatedRedirection = ({
     // once user has successfully logged in. Otherwise, redirect to the specified redirect url.
     // Note: For multiple enterprise use case, we need to make sure that user first visits the
     // enterprise selection page and then complete the auth workflow
-    if (finishAuthUrl && !redirectUrl.includes(finishAuthUrl)) {
+    if (redirectToRecommendationsPage) {
+      const recommendationsUrl = new URL(`${getConfig().AUTH_BASE_URL}${AUTH_RECOMMENDATIONS_URL}`);
+      const searchParams = new URLSearchParams();
+
+      searchParams.set('country', country);
+      if (educationLevel) {
+        searchParams.set('levelOfEducation', educationLevel);
+      }
+      if (finishAuthUrl) {
+        searchParams.set('finishAuthUrl', finishAuthUrl);
+      }
+
+      recommendationsUrl.search = searchParams.toString();
+      finalRedirectUrl = recommendationsUrl.toString();
+    } else if (finishAuthUrl && !redirectUrl.includes(finishAuthUrl)) {
       finalRedirectUrl = getConfig().LMS_BASE_URL + finishAuthUrl;
     } else {
       finalRedirectUrl = redirectUrl;
```

Comment:
this would not handle the next URL and will only redirect the users to dashboard after seeing recommendations. 
We should first prepare the finalRedirectUrl as it is and should not change its implementation
```
if (finishAuthUrl && !redirectUrl.includes(finishAuthUrl)) {
finalRedirectUrl = getConfig().LMS_BASE_URL + finishAuthUrl;
    } else {
      finalRedirectUrl = redirectUrl;
```

then after this code we can have check for recommendations as we have for progressive profiling

```
f (redirectToRecommendationsPage) {
      const recommendationsUrl = new URL(`${getConfig().AUTH_BASE_URL}${AUTH_RECOMMENDATIONS_URL}`);
      const searchParams = new URLSearchParams();

      searchParams.set('country', country);
      if (educationLevel) {
        searchParams.set('levelOfEducation', educationLevel);
      }
      if (finishAuthUrl) {
        searchParams.set('finishAuthUrl', finishAuthUrl);
      }

      recommendationsUrl.search = searchParams.toString();
      finalRedirectUrl = recommendationsUrl.toString();
    } 
```
 
 we should change this above code to retain all the query params available plus should pass the finalRedirectUrl instead of finishAuth Url in query param so that recommendations page can redirect to that url.